terraform/modules/monitoring_secrets/kms.tf

```diff
@@ -1,27 +1,19 @@
 resource "aws_kms_key" "this" {
-  count = var.create_email_subscription ? 1 : 0
```

Comment:
Everything in the module ended up with this condition, so if not wanted the entrypoint should just not call the module.